### ISSUE TITLE
Cannot downgrade to basic type checking when set to strict

### ISSUE THREAD

**Issue by linter-purgatory** [microsoft/pyright#4751]
**Describe the bug**
We have a project with `typeCheckingMode="strict"` in `pyproject.toml`. For some files, we wish to downgrade it to basic type checking mode, see below for reasoning. We put a `# pyright: basic` comment at the beginning of the file, but the file continues to be checked in strict mode.

**To Reproduce**
note: pushed a repo that illustrates the issue with pyright `1.1.298`: https://github.com/linter-purgatory/pyright-issue

1. Create a file with something that is an error in strict and is fine in basic
2. Create `pyproject.toml` with `typeCheckingMode="strict"`
3. Run `npx pyright`, you get errors when there should be none.

**Expected behavior**
Basic type checking mode should be enabled when the comment is present, regardless of the overall configuration.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
# pyright: basic,reportUnknownLambdaType=warning

def fn(unknown_variable_type): # this should be surpressed in basic mode, but isn't
    print(unknown_variable_type)

    lambda y: 1 # this error is correctly turned into a warning via the comment

```

**VS Code extension or command-line**
Occurs in cli @ `1.1.298`, as well as pylance `v2023.3.10` (ie in the editor). I don't have the extension installed.

**Additional context**
- The inverse works - we can upgrade from basic to strict with a comment
- If we specify additional diagnostic settings, eg `# pyright: strict,reportUnknownLambdaType=warning`, the additional settings are always respected

We have some files that use pandas, which generates a lot of spurious errors in strict mode because the the typing for pandas is incomplete, eg `error: Type of "map" is unknown (reportUnknownMemberType)` on `DataFrame['column].map`. I don't think this is an issue with pyright, but if you have any tips they'd be appreciated.

**Comment by erictraut** [comment]
This is by design. If your config file indicates that a file should be strict, it can't be downgraded to basic in the file.

If your project is not ready for strict type checking across the board, you should set your project to basic type checking and mark individual files or directories as strict.

**Comment by erictraut** [comment]
When I originally designed the `# pyright` comment mechanism, I had a good reason to not permit "downgrading" files that were marked as "strict". At that time, I was leading a largish team of developers, and I wanted to prevent individual team members from disabling strict type checking on files that had already been marked strict. This was the best way to make continued progress toward better static typing. I didn't have the bandwidth to monitor every change to every file made by my team, but I was able to monitor the `pyrightconfig.json` file (which was modified infrequently). I installed a commit notification on the pyrightconfig.json file and reviewed all such changes to make sure that we weren't reverting existing "strict" files or directories.

This was long before I added the `# pyright: basic` ability. I can see why it would be natural to expect `# pyright: basic` to override a "strict" setting in the config — especially given that you can "downgrade" individual diagnostic rules using file-local comments (e.g. `reportMissingSource=false`). I see how the current behavior is confusing and arguably inconsistent. Thinking about this more, I think it's worth changing the intended design.

**Comment by erictraut** [comment]
One open question is whether `# pyright: basic` should revert back to all of the "basic type checking mode" defaults or whether it should revert back to the severity levels specified in your config & settings. I'm thinking the former, but I could make a good argument for the latter as well.

**Comment by erictraut** [comment]
I went with the former approach and documented it. This will be included in the next release.

**Comment by linter-purgatory** [comment]
You're fast! This fixes our problem here - thanks.

I'm not sure if this is what you meant by option (2) though, but for what its worth: since "basic" and "strict" are kind of the base that the config file and comments apply changes to, you could expect `# pyright: basic` to change that base but still apply any customizations to specific diagnostic settings in the config files or comments (in my mind, it's just overriding that particular entry in the config file, and would expect it to behave as if I had changed it there)

I think I've crossed into splitting hairs territory though, and I think what you've chosen is also perfectly reasonable.

**Comment by erictraut** [comment]
The problem is that the "base" is defined by the `typeCheckingMode` set in your config file or language server settings. So if you set "typeCheckingMode" to "off", then the base set of rules will be all off, and if you set "typeCheckingMode" to "strict", then all of the strict rules will be on. It wouldn't make sense to adopt these "base" rules in a file when you specify "# pyright: basic". I think that would be counterintuitive to most users. It makes more sense to adopt the "basic" rule set defaults, which is what I've implemented.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.299, which I just published. It will also be included in a future release of pylance.
